ultralytics/utils/metrics.py

```diff
@@ -505,6 +505,29 @@ def print(self):
         for i in range(self.matrix.shape[0]):
             LOGGER.info(" ".join(map(str, self.matrix[i])))
 
+    def summary(self, **kwargs):
+        """
+        Returns summary of the confusion matrix for export in different formats CSV, XML, HTML.
+
+        Each row represents predicted class and columns represent actual class.
+        """
+        import re
+
+        names = self.names if self.task == "classify" else self.names + ["background"]  # background class for 'detect'
+
+        clean_names = []
+        for name in names:  # Clean names for valid XML and SQL usage
+            name = re.sub(r"[^a-zA-Z0-9_]", "_", name)
+            if name == "cardigan":  # Handle imagenet10 dataset conflict between 'cardigan' and 'Cardigan' for to_sql().
```

Comment:
@RizwanMunawar this is strangely non-robust to any other datasets. I think we shouldn't have carve-outs that are specific to one thing, it's going to lead to very messy code. 

If there's a bug it must be resolved in a robust way, not a super-special set of code for ImageNet.